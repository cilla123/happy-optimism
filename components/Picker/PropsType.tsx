import { BasePickerViewProps } from '../PickerView/PropsType';

export interface BasePickerProps extends BasePickerViewProps {
  visible?: boolean;
  title?: string;
  okText?: string;
  cancelText?: string;
  onOk?: (value?: object) => void;
  onCancel?: () => void;
  onMaskClick?: () => void;
  children?: any;
  onTransition?: (value: boolean) => void;
}
