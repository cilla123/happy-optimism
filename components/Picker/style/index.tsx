import '../../style/';
import '../../Popup/style/';
import '../../PickerView/style';
import './index.scss';
