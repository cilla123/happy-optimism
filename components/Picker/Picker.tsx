import React, { PureComponent } from 'react';
import classnames from 'classnames';
import Popup from '../Popup';
import PickerView from '../PickerView';
import { BasePickerProps } from './PropsType';

const getValue = (props, defaultValue?: any) => {
  if ('value' in props && props.value.length > 0) {
    return [].concat(props.value);
  }

  if ('defaultValue' in props && props.defaultValue.length > 0) {
    return [].concat(props.defaultValue);
  }

  return defaultValue;
};

export interface PickerProps extends BasePickerProps {
  prefixCls?: string;
  className?: any;
}

export default class Picker extends PureComponent<PickerProps, any> {

  static Stack: any;

  static defaultProps = {
    title: '请选择',
    cancelText: '取消',
    okText: '确定',
    dataSource: [],
    prefixCls: 'za-picker',
    valueMember: 'value',
    itemRender: data => data.label,
  };

  private tempValue;
  private tempObjValue;
  private isScrolling;

  constructor(props) {
    super(props);
    this.state = {
      visible: props.visible || false,
      value: getValue(props, []),
      objValue: [],
    };

    this.tempValue = this.state.value;
    this.tempObjValue = this.state.objValue;
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      visible: nextProps.visible,
      value: getValue(nextProps, []),
    });
  }

  onInit = (selected) => {
    const { valueMember, onInit } = this.props;
    this.setState({
      firstValue: selected.map(item => item[valueMember!]),
      firstObjValue: selected,
    });

    if (typeof onInit === 'function') {
      onInit(selected);
    }
  }

  onChange = (selected) => {
    const { valueMember, onChange } = this.props;
    const value = selected.map(item => item[valueMember!]);
    this.setState({
      value,
      objValue: selected,
    });

    if (typeof onChange === 'function') {
      onChange(selected);
    }
  }

  onCancel = () => {
    const { onCancel } = this.props;
    this.toggle();
    this.setState({
      value: this.tempValue,
      objValue: this.tempObjValue,
    });
    if (typeof onCancel === 'function') {
      onCancel();
    }
  }

  onOk = () => {
    if (this.isScrolling) {
      return false;
    }
    const value = this.state.value.length === 0 ? this.state.firstValue : this.state.value;
    const objValue = this.state.objValue.length === 0 ? this.state.firstObjValue : this.state.objValue;

    this.setState({
      value,
      objValue,
    });

    const { onOk } = this.props;
    if (typeof onOk === 'function') {
      onOk(objValue);
    }
    this.toggle();
  }

  onMaskClick = () => {
    const { onMaskClick } = this.props;
    this.onCancel();
    if (typeof onMaskClick === 'function') {
      onMaskClick();
    }
  }

  // 切换显示状态
  toggle = (visible = false) => {
    this.setState({ visible });
  }

  onTransition(isScrolling) {
    this.isScrolling = isScrolling;
  }

  render() {
    const { prefixCls, className, cancelText, okText, title, children, ...others } = this.props;
    const { visible, value, firstObjValue } = this.state;

    const cls = classnames(prefixCls, className);
    // const content = children && cloneElement(children, {
    //   onClick: () => this.toggle(true),
    // });

    return (
      <div className={cls}>
      <Popup
        visible={visible}
        onMaskClick={this.onMaskClick}
      >
        <div className={`${prefixCls}-wrapper`}>
          <div className={`${prefixCls}-header`}>
            <div className={`${prefixCls}-cancel`} onClick={this.onCancel}>{cancelText}</div>
            <div className={`${prefixCls}-title`}>{title}</div>
            <div className={`${prefixCls}-submit`} onClick={this.onOk}>{okText}</div>
          </div>
          <PickerView
            {...others}
            prefixCls={prefixCls}
            value={value}
            firstObjValue={firstObjValue}
            onInit={this.onInit}
            onChange={this.onChange}
            onTransition={(isScrolling) => { this.onTransition(isScrolling); }}
          />
        </div>
      </Popup>
      </div>
    );
  }
}
