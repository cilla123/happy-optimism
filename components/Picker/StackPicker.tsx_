import React, { Component } from 'react';
import classnames from 'classnames';
import Popup from '../Popup';
import { BasePickerProps } from './PropsType';

const stopEventPropagation = (e) => {
  e.stopPropagation();
  // e.nativeEvent.stopImmediatePropagation();
};

const labelAddon = ' > ';
const displayItems = 8;
const itemHeight = 50;
export interface PickerStackProps extends BasePickerProps {
  prefixCls?: string;
  className?: any;
}

export default class PickerStack extends Component<PickerStackProps, any> {

  static defaultProps = {
    prefixCls: 'za-picker',
    value: [],
    valueMember: 'value',
    title: '请选择',
    placeholder: '请选择',
    disabled: false,
    dataSource: [],
    cols: Infinity,
    // displayItems: 8,
    // itemHeight: 50,
    displayRender: data => data.map(({ label }) => label).join(''),
    itemRender: data => data.label,
    validate: () => {},
    onOk: () => {},
    onCancel: () => {},
  };

  private columns;

  constructor(props) {
    super(props);
    this.columns = {};
    this.state = {
      visible: false,
      ...this.resolveProps(props),
    };
  }

  componentDidMount() {
    this.reposition();
  }

  componentWillReceiveProps(nextProps) {
    this.setState(this.resolveProps(nextProps));
  }

  componentDidUpdate() {
    this.reposition();
  }

  onMaskClick = () => {
    const { onMaskClick } = this.props;
    this.close(true);
    if (typeof onMaskClick === 'function') {
      onMaskClick();
    }
  }

  onCancel = () => {
    const { onCancel } = this.props;
    this.close(true);
    if (typeof onCancel === 'function') {
      onCancel();
    }
  }

  resolveProps = ({ value, dataSource, validate }) => {
    const resolveValue: any[] = [];

    value.reduce((list, item) => {
      const valueItem = this.obtainItem(list, item);

      if (valueItem) {
        resolveValue.push(valueItem);

        return valueItem.children || [];
      }

      return [];
    }, dataSource);

    return {
      value: resolveValue,
      displayValue: [...resolveValue],
      errorMsg: validate(value) || '',
    };
  }

  obtainItem = (list, value) => {
    const { valueMember = PickerStack.defaultProps.valueMember } = this.props;
    return list.filter(item => item[valueMember] === value)[0];
  }

  change = (index: number, cVal?: object, isLast?: boolean) => {
    // const { onOk } = this.props;
    const value = this.state.value.slice(0, index);
    let errorMsg: string = '';

    if (cVal) {
      value[index] = cVal;
    }
    // errorMsg = validate(value);

    if (isLast && !errorMsg) {
      this.setState({
        value,
        displayValue: [...value],
        errorMsg,
      });
      this.close();
      // onOk(value);
    } else {
      this.setState({ value, errorMsg });
    }
  }

  show = () => {
    const { disabled } = this.state;
    if (!disabled) {
      this.setState({ visible: true });
    }
  }

  close = (isCancel?: boolean) => {
    this.setState({ visible: false });
    if (isCancel) {
      this.setState({
        value: [...this.state.displayValue],
      });
    }
  }

  reposition = () => {
    const { dataSource, valueMember, cols } = this.props;
    this.state.value.reduce((data, item, index) => {
      const value = item[valueMember!];
      const valIndex = data.map(dataItem => dataItem[valueMember!]).indexOf(value);

      if (index < cols! && valIndex !== -1) {
        const target = this.columns[`column${index}`];
        const position = target.scrollTop;
        const viewTopIndex = valIndex - displayItems;

        if (position < ((viewTopIndex + 1) * itemHeight) || position > (valIndex * itemHeight)) {
          target.scrollTop = valIndex * itemHeight;
        }

        return data[valIndex].children || [];
      }

      return [];
    }, dataSource);
  }

  renderGroup(dataSource, value) {
    const { valueMember = PickerStack.defaultProps.valueMember, cols = PickerStack.defaultProps.cols } = this.props;
    const group: any[] = [];
    let i = 0;

    while (dataSource) {
      const colVal = value[i];
      const childrenData = ((colVal ? this.obtainItem(dataSource, colVal[valueMember]) : dataSource[0]) || {}).children;

      if (childrenData && childrenData.length && (i < (cols - 1))) {
        group.push(this.renderColumn(dataSource, i, colVal || { [valueMember]: '' }));
        dataSource = childrenData;
      } else {
        group.push(this.renderColumn(dataSource, i, colVal || { [valueMember]: '' }, true));
        dataSource = null;
      }

      i += 1;
    }

    return group;
  }

  renderColumnItem = (list, colIndex, data, isLast): React.ReactNode => {
    const { valueMember = PickerStack.defaultProps.valueMember, prefixCls,
      itemRender = PickerStack.defaultProps.itemRender } = this.props;
    const pickVal = data[valueMember];
    return list.map((item, index) => {
      return (
        <div
          key={+index}
          className={classnames(`${prefixCls}-stack-item`, { active: item[valueMember] === pickVal })}
          onClick={() => this.change(colIndex, item, isLast)}
        >
          {itemRender(item)}
        </div>
      );
    });
  }

  renderColumn = (list: any[], colIndex: number, data: object, isLast?: boolean) => {
    const { valueMember = PickerStack.defaultProps.valueMember, prefixCls } = this.props;
    const pickVal = data[valueMember];
    const cls = classnames(`${prefixCls}-stack-column`, {
      'lower-hidden': !pickVal,
    });

    return (
      <div
        key={colIndex}
        className={cls}
        onClick={() => this.change(colIndex - 1)}
      >
        <div
          className={`${prefixCls}-stack-column-wrapper`}
          ref={(ref) => { this.columns[`column${colIndex}`] = ref; }}
          onClick={stopEventPropagation}
        >
        {this.renderColumnItem(list, colIndex, data, isLast)}
        </div>
      </div>
    );
  }

  renderWrapper() {
    const { title, dataSource, prefixCls, itemRender } = this.props;
    const { visible, errorMsg, value } = this.state;

    return (
      <Popup
        visible={visible}
        onMaskClick={this.onMaskClick}
      >
        <div className={`${prefixCls}-wrapper`}>
          <div className={`${prefixCls}-header`}>
            <div className={`${prefixCls}-cancel`} onClick={this.onCancel}>取消</div>
            <div className={`${prefixCls}-title`}>{title}</div>
            <div className={`${prefixCls}-submit`} />
          </div>
          <div className={`${prefixCls}-crumbs`}>
            <p>选择：{value.map(item => itemRender!(item)).join(labelAddon)}</p>
            {errorMsg ? <p className={`${prefixCls}-crumbs-error`}>{errorMsg}</p> : null}
          </div>
          <div className={`${prefixCls}-stack-group`}>{this.renderGroup(dataSource, value)}</div>
        </div>
      </Popup>
    );
  }

  render() {
    const { prefixCls, valueMember } = this.props;
    const { displayValue } = this.state;
    let displayLabel = displayValue.map(({ label }) => label).join('');

    const labelCls = classnames(`${prefixCls}-input`, {
      [`${prefixCls}-placeholder`]: !displayLabel,
    });

    return (
      <div className={prefixCls}>
        <div
          className={labelCls}
          onClick={this.show}
        >
          <input type="hidden" value={displayValue.map(data => data[valueMember!]).join(',')} />
          {displayLabel}
        </div>
        {this.renderWrapper()}
      </div>
    );
  }
}
