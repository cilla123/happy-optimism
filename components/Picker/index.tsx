import Picker from './Picker';
// import Stack from './StackPicker';

// Picker.Stack = Stack;

export default Picker;
