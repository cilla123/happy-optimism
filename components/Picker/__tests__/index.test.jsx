import React from 'react';
import { render, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Picker from '../index';

const District = [
  {
    value: '1',
    label: '北京市',
    children: [
      { value: '11', label: '海淀区' },
      { value: '12', label: '西城区' },
    ],
  },
  {
    value: '2',
    label: '上海市',
    children: [
      { value: '21', label: '杨浦区' },
      { value: '22', label: '静安区' },
    ],
  },
];


describe('Picker', () => {
  it('Picker render visible', () => {
    const wrapper = mount(
      <Picker
        prefixCls="za-picker"
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        visible
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render custom label', () => {
    const wrapper = mount(
      <Picker
        dataSource={[
          {
            code: '1',
            name: '北京市',
            children: [
              { code: '11', name: '海淀区' },
              { code: '12', name: '西城区' },
            ],
          },
          {
            code: '2',
            name: '上海市',
            children: [
              { code: '21', name: '黄埔区' },
              { code: '22', name: '虹口区' },
            ],
          },
        ]}
        valueMember="code"
        itemRender={data => data.name}
        />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('should trigger onOk when press ok button', () => {
  //   const onOkFn = jest.fn();
  //   const onCancelFn = jest.fn();

  //   const wrapper = mount(
  //     <Picker
  //       dataSource={[
  //         {
  //           value: '1',
  //           label: '选项一',
  //           children: [
  //             { value: '11', label: '选项一' },
  //             { value: '12', label: '选项二' },
  //           ],
  //         },
  //         {
  //           value: '2',
  //           label: '选项一',
  //           children: [
  //             { value: '21', label: '选项一' },
  //             { value: '22', label: '选项二' },
  //           ],
  //         },
  //       ]}
  //       value={['1', '12']}
  //       onOk={onOkFn}
  //       onCancel={onCancelFn}
  //       />
  //   );

  //   wrapper.find('.za-picker-submit').simulate('click');
  //   wrapper.find('.za-picker').simulate('click');
  //   expect(onOkFn).toBeCalled();
  //   expect(onCancelFn).not.toBeCalled();
  // });

  // it('should trigger onCancel when press cancel button', () => {
  //   const onOkFn = jest.fn();
  //   const onCancelFn = jest.fn();

  //   const wrapper = mount(
  //     <Picker
  //       dataSource={[
  //         {
  //           value: '1',
  //           label: '选项一',
  //           children: [
  //             { value: '11', label: '选项一' },
  //             { value: '12', label: '选项二' },
  //           ],
  //         },
  //         {
  //           value: '2',
  //           label: '选项一',
  //           children: [
  //             { value: '21', label: '选项一' },
  //             { value: '22', label: '选项二' },
  //           ],
  //         },
  //       ]}
  //       value={['1', '12']}
  //       onOk={onOkFn}
  //       onCancel={onCancelFn}
  //       />
  //   );

  //   wrapper.find('.za-picker-cancel').simulate('click');
  //   expect(onCancelFn).toBeCalled();
  //   expect(onOkFn).not.toBeCalled();
  // });

  it('receive new dataSource', () => {
    const wrapper = shallow(
      <Picker
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    wrapper.setProps({
      dataSource: [
        { value: 'a', label: '选项一' },
        { value: 'b', label: '选项二' },
        { value: 'c', label: '选项三' },
      ],
    });
  });

  it('receive new value', () => {
    const wrapper = shallow(
      <Picker
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    wrapper.setProps({ value: '1' });
  });

  it('receive new cascader dataSource', () => {
    const wrapper = shallow(
      <Picker
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        />
    );

    wrapper.setProps({
      dataSource: [
        {
          value: '3',
          label: '选项一',
          children: [
            { value: '31', label: '选项一' },
            { value: '32', label: '选项二' },
          ],
        },
        {
          value: '4',
          label: '选项一',
          children: [
            { value: '41', label: '选项一' },
            { value: '42', label: '选项二' },
          ],
        },
      ],
    });
  });

  // it('StackPicker', () => {
  //   jest.useFakeTimers();
  //   const wrapper = mount(
  //     <Picker.Stack
  //       dataSource={District}
  //       />
  //   );
  //   expect(toJson(wrapper)).toMatchSnapshot();
  //   wrapper.setProps({ value: ['安徽省', '安庆市', '大观区'] });
  //   jest.runAllTimers();
  //   wrapper.unmount();
  // });

  // it('StackPicker init value', () => {
  //   const wrapper = mount(
  //     <Picker.Stack
  //       dataSource={District}
  //       value={['340000', '340800', '340803']}
  //       />
  //   );
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });

  // it('StackPicker changeValue', () => {
  //   const onChangeFn = jest.fn();
  //   const wrapper = mount(
  //     <Picker.Stack
  //       dataSource={District}
  //       onOk={onChangeFn}
  //       />
  //   );

  //   wrapper.find('.za-picker-stack-column').at(0).simulate('click');
  //   wrapper.find('.za-picker-stack-item').at(0).simulate('click');
  //   expect(onChangeFn).toBeCalled();
  // });

  // it('StackPicker maskClick', () => {
  //   const wrapper = mount(
  //     <Picker.Stack
  //       dataSource={District}
  //       />
  //   );
  //   wrapper.find('.za-picker-input').simulate('click');
  //   wrapper.find('.za-picker-cancel').simulate('click');
  //   wrapper.find('.za-mask').simulate('click');
  // });
});
