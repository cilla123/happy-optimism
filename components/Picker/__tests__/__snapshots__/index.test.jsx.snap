// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Picker render visible 1`] = `
<Picker
  cancelText="取消"
  dataSource={
    Array [
      Object {
        "label": "选项一",
        "value": "1",
      },
      Object {
        "label": "选项二",
        "value": "2",
      },
    ]
  }
  itemRender={[Function]}
  okText="确定"
  prefixCls="za-picker"
  title="请选择"
  valueMember="value"
  visible={true}
>
  <div
    className="za-picker"
  >
    <Popup
      animationDuration={200}
      autoClose={false}
      direction="bottom"
      mask={true}
      maskType="normal"
      onMaskClick={[Function]}
      prefixCls="za-popup"
      stayTime={3000}
      visible={true}
    >
      <div>
        <Portal
          getContainer={[Function]}
        >
          <div
            className="za-popup"
          >
            <div
              className="za-popup-wrapper za-popup-wrapper-bottom"
              style={
                Object {
                  "WebkitTransitionDuration": "200ms",
                  "transitionDuration": "200ms",
                }
              }
            >
              <div
                className="za-picker-wrapper"
              >
                <div
                  className="za-picker-header"
                >
                  <div
                    className="za-picker-cancel"
                    onClick={[Function]}
                  >
                    取消
                  </div>
                  <div
                    className="za-picker-title"
                  >
                    请选择
                  </div>
                  <div
                    className="za-picker-submit"
                    onClick={[Function]}
                  >
                    确定
                  </div>
                </div>
                <PickerView
                  cols={Infinity}
                  dataSource={
                    Array [
                      Object {
                        "label": "选项一",
                        "value": "1",
                      },
                      Object {
                        "label": "选项二",
                        "value": "2",
                      },
                    ]
                  }
                  disabled={false}
                  firstObjValue={
                    Array [
                      Object {
                        "label": "选项一",
                        "value": "1",
                      },
                    ]
                  }
                  itemRender={[Function]}
                  onChange={[Function]}
                  onInit={[Function]}
                  onTransition={[Function]}
                  prefixCls="za-picker"
                  value={Array []}
                  valueMember="value"
                  visible={true}
                >
                  <div
                    className="za-picker-panel"
                  >
                    <div
                      className="za-picker-mask-top"
                    />
                    <div
                      className="za-picker-view"
                    >
                      <Wheel
                        dataSource={
                          Array [
                            Object {
                              "label": "选项一",
                              "value": "1",
                            },
                            Object {
                              "label": "选项二",
                              "value": "2",
                            },
                          ]
                        }
                        disabled={false}
                        itemRender={[Function]}
                        key="0"
                        onChange={[Function]}
                        onTransition={[Function]}
                        prefixCls="za-wheel"
                        value="1"
                        valueMember="value"
                      >
                        <div
                          className="za-wheel"
                        >
                          <div
                            className="za-wheel-content"
                          >
                            <div
                              className="za-wheel-item za-wheel-item-selected"
                              key="0"
                            >
                              选项一
                            </div>
                            <div
                              className="za-wheel-item"
                              key="1"
                            >
                              选项二
                            </div>
                          </div>
                        </div>
                      </Wheel>
                    </div>
                    <div
                      className="za-picker-mask-bottom"
                    />
                  </div>
                </PickerView>
              </div>
            </div>
            <Mask
              className="fade-enter"
              onClose={[Function]}
              prefixCls="za-mask"
              style={
                Object {
                  "WebkitAnimationDuration": "200ms",
                  "animationDuration": "200ms",
                }
              }
              type="normal"
              visible={true}
            >
              <div
                className="za-mask fade-enter normal"
                onClick={[Function]}
                style={
                  Object {
                    "WebkitAnimationDuration": "200ms",
                    "animationDuration": "200ms",
                  }
                }
              />
            </Mask>
          </div>
        </Portal>
      </div>
    </Popup>
  </div>
</Picker>
`;

exports[`Picker render custom label 1`] = `
<Picker
  cancelText="取消"
  dataSource={
    Array [
      Object {
        "children": Array [
          Object {
            "code": "11",
            "name": "海淀区",
          },
          Object {
            "code": "12",
            "name": "西城区",
          },
        ],
        "code": "1",
        "name": "北京市",
      },
      Object {
        "children": Array [
          Object {
            "code": "21",
            "name": "黄埔区",
          },
          Object {
            "code": "22",
            "name": "虹口区",
          },
        ],
        "code": "2",
        "name": "上海市",
      },
    ]
  }
  itemRender={[Function]}
  okText="确定"
  prefixCls="za-picker"
  title="请选择"
  valueMember="code"
>
  <div
    className="za-picker"
  >
    <Popup
      animationDuration={200}
      autoClose={false}
      direction="bottom"
      mask={true}
      maskType="normal"
      onMaskClick={[Function]}
      prefixCls="za-popup"
      stayTime={3000}
      visible={false}
    >
      <div>
        <Portal
          getContainer={[Function]}
        >
          <div
            className="za-popup za-popup-hidden"
          >
            <div
              className="za-popup-wrapper za-popup-wrapper-bottom"
              style={
                Object {
                  "WebkitTransitionDuration": "200ms",
                  "transitionDuration": "200ms",
                }
              }
            >
              <div
                className="za-picker-wrapper"
              >
                <div
                  className="za-picker-header"
                >
                  <div
                    className="za-picker-cancel"
                    onClick={[Function]}
                  >
                    取消
                  </div>
                  <div
                    className="za-picker-title"
                  >
                    请选择
                  </div>
                  <div
                    className="za-picker-submit"
                    onClick={[Function]}
                  >
                    确定
                  </div>
                </div>
                <PickerView
                  cols={Infinity}
                  dataSource={
                    Array [
                      Object {
                        "children": Array [
                          Object {
                            "code": "11",
                            "name": "海淀区",
                          },
                          Object {
                            "code": "12",
                            "name": "西城区",
                          },
                        ],
                        "code": "1",
                        "name": "北京市",
                      },
                      Object {
                        "children": Array [
                          Object {
                            "code": "21",
                            "name": "黄埔区",
                          },
                          Object {
                            "code": "22",
                            "name": "虹口区",
                          },
                        ],
                        "code": "2",
                        "name": "上海市",
                      },
                    ]
                  }
                  disabled={false}
                  firstObjValue={
                    Array [
                      Object {
                        "code": "1",
                        "name": "北京市",
                      },
                      Object {
                        "code": "11",
                        "name": "海淀区",
                      },
                    ]
                  }
                  itemRender={[Function]}
                  onChange={[Function]}
                  onInit={[Function]}
                  onTransition={[Function]}
                  prefixCls="za-picker"
                  value={Array []}
                  valueMember="code"
                >
                  <div
                    className="za-picker-panel"
                  >
                    <div
                      className="za-picker-mask-top"
                    />
                    <div
                      className="za-picker-view"
                    >
                      <Wheel
                        dataSource={
                          Array [
                            Object {
                              "code": "1",
                              "name": "北京市",
                            },
                            Object {
                              "code": "2",
                              "name": "上海市",
                            },
                          ]
                        }
                        disabled={false}
                        itemRender={[Function]}
                        key="0"
                        onChange={[Function]}
                        onTransition={[Function]}
                        prefixCls="za-wheel"
                        value="1"
                        valueMember="code"
                      >
                        <div
                          className="za-wheel"
                        >
                          <div
                            className="za-wheel-content"
                          >
                            <div
                              className="za-wheel-item za-wheel-item-selected"
                              key="0"
                            >
                              北京市
                            </div>
                            <div
                              className="za-wheel-item"
                              key="1"
                            >
                              上海市
                            </div>
                          </div>
                        </div>
                      </Wheel>
                      <Wheel
                        dataSource={
                          Array [
                            Object {
                              "code": "11",
                              "name": "海淀区",
                            },
                            Object {
                              "code": "12",
                              "name": "西城区",
                            },
                          ]
                        }
                        disabled={false}
                        itemRender={[Function]}
                        key="1"
                        onChange={[Function]}
                        onTransition={[Function]}
                        prefixCls="za-wheel"
                        value="11"
                        valueMember="code"
                      >
                        <div
                          className="za-wheel"
                        >
                          <div
                            className="za-wheel-content"
                          >
                            <div
                              className="za-wheel-item za-wheel-item-selected"
                              key="0"
                            >
                              海淀区
                            </div>
                            <div
                              className="za-wheel-item"
                              key="1"
                            >
                              西城区
                            </div>
                          </div>
                        </div>
                      </Wheel>
                    </div>
                    <div
                      className="za-picker-mask-bottom"
                    />
                  </div>
                </PickerView>
              </div>
            </div>
            <Mask
              className="fade-leave"
              onClose={[Function]}
              prefixCls="za-mask"
              style={
                Object {
                  "WebkitAnimationDuration": "200ms",
                  "animationDuration": "200ms",
                }
              }
              type="normal"
              visible={false}
            />
          </div>
        </Portal>
      </div>
    </Popup>
  </div>
</Picker>
`;
